examples/experimental/horizontal-pod-autoscaling/README.md

```diff
@@ -0,0 +1,207 @@
+
+
+## Step One: Run & expose php-apache server
+
+To demonstrate horizontal pod autoscaler we will use a custom docker image based on php-apache server.
+The image can be found [here](image/).
+It defines [index.php](image/index.php) page which performs some CPU intensive computations.
+
+First, we will start a replication controller running the image and expose it as an external service:
+```console
+$ kubectl create -f ./examples/experimental/horizontal-pod-autoscaling/rc-php-apache.yaml
+replicationcontrollers/php-apache
+
+$ kubectl expose rc php-apache --port=80 --type=LoadBalancer
+NAME         LABELS           SELECTOR         IP(S)     PORT(S)
+php-apache   run=php-apache   run=php-apache             80/TCP
+```
+
+Now, we will wait some time and verify that both the replication controller and the service were correctly created and are running. We will also determine the IP address of the service:
+
+```console
+$ kubectl get pods
+NAME               READY     STATUS    RESTARTS   AGE
+php-apache-wa3t1   1/1       Running   0          12m
+
+$ kubectl describe services php-apache | grep "LoadBalancer Ingress"
+LoadBalancer Ingress:	146.148.24.244
+```
+
+We may now check that php-apache server works correctly by calling ``curl`` with the service's IP:
+
+```console
+$ curl http://146.148.24.244
+OK!
+```
+
+Please notice that when exposing the service we assumed that our cluster runs on a provider which supports load balancers (e.g.: on GCE).
+If load balancers are not supported (e.g.: on Vagrant), we can expose php-apache service as ``ClusterIP`` and connect to it using the proxy on the master:
+
+```console
+$ kubectl expose rc php-apache --port=80 --type=ClusterIP
+NAME         LABELS           SELECTOR         IP(S)     PORT(S)
+php-apache   run=php-apache   run=php-apache             80/TCP
+
+$ kubectl cluster-info | grep master
+Kubernetes master is running at https://146.148.6.215
+
+$ curl -k -u admin:password https://146.148.6.215/api/v1/proxy/namespaces/default/services/php-apache/
+OK!
+```
+
+
+## Step Two: Create horizontal pod autoscaler
+
+Now that the server is running, we will create a horizontal pod autoscaler for it.
+To create it, we will use [hpa-php-apache.yaml](hpa-php-apache.yaml) file, which look like this:
+
+```yaml
+apiVersion: experimental/v1alpha1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: php-apache
+  namespace: default
+spec:
+  maxReplicas: 10
+  minReplicas: 1
+  scaleRef:
+    kind: ReplicationController
+    name: php-apache
+    namespace: default
+  target:
+    quantity: 100m
+    resource: cpu
+```
+
+Please be aware that it sets the target CPU consumption to 100 milli-cores, while in [rc-php-apache.yaml](rc-php-apache.yaml) each pod requests 200 milli-cores.
+As a general rule, the autoscaler's target should be lower than the request.
+Otherwise, overloaded pods may not be able to consume more resources than requested what will prevent scaling-up.
```

Comment:
I would rewrite this sentence to be a little clearer: "Otherwise, overloaded pods may not be able to consume more than the autoscaler's target utilization, thereby preventing the autoscaler from seeing high enough utilization to trigger it to scale up."